noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -560,43 +637,123 @@ comptime fn generate_finalization_payload(
     };
 
     // We compute the log length and we concatenate the fields into a single quote.
-    let log_length = fields_list.len();
+    let public_values_length = fields_list.len();
     let fields = fields_list.join(quote {,});
 
     // Now we compute quotes relevant to the multi-scalar multiplication.
-    let (generators_list, scalars_list, args_list, msm_aux_vars) =
+    let (generators_list, _, args_list, msm_aux_vars) =
         generate_multi_scalar_mul(indexed_nullable_fields);
 
+    // We generate scalars_list manually as we need it to refer self.public_values
+    let mut scalars_list: [Quoted] = &[];
+    for i in 0..public_values_length {
+        scalars_list =
+            scalars_list.push_back(quote { std::hash::from_field_unsafe(self.public_values[$i]) });
+    }
+
     let generators = generators_list.join(quote {,});
     let scalars = scalars_list.join(quote {,});
     let args = args_list.join(quote {,});
 
+    // Then we compute values for `encrypt_log(...)` function
+    let setup_log_plaintext_length = indexed_fixed_fields.len() * 32 + 64;
+    // 353 is a fixed ciphertext overhead as defined by `compute_private_log_payload`
+    let setup_log_byte_length = 353 + setup_log_plaintext_length;
```

Comment:
This value shows up a bunch, might make sense to put it in some constants file.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/9356/commits/1c7b07aa6ff90ca913164e254f78ddedc8028267